### ISSUE TITLE
Generator Loss Function

### ISSUE THREAD

**Issue by darylfung96** [yunjey/StarGAN#26]
In the paper it says, Generator loss function is 
<img width="394" alt="screen shot 2018-01-28 at 3 59 40 pm" src="https://user-images.githubusercontent.com/24358642/35487701-51fb70ec-0444-11e8-88c9-7810f94ae66b.png">

But in the code, it was `g_loss = g_loss_fake + self.lambda_rec * g_loss_rec + self.lambda_cls * g_loss_cls`

Why is g_loss_real excluded in this part?

**Comment by yunjey** [comment]
@darylfung96 Let `loss_adv = loss_real + loss_fake`. Because **G** is not used in computing `loss_real`, minimizing `loss_fake` with respect to **G** is equivalent to minimizing `loss_real + loss_fake` with respect to **G**. We excluded the `loss_real ` part when optimizing the generator G to eliminate unnecessary operations. This is not the technique used only in StarGAN paper, but the way most GAN papers use.

**Comment by darylfung96** [comment]
Thank you!

**Comment by tsly123** [comment]
Hi @yunjey ,

Why did you determine the lambda 1 and 10 for classification and reconstruction loss, respectively?
Thank you
